### PR TITLE
ci(github): update coverage action version

### PR DESCRIPTION
**Pull request by Ja7ad**
## Description

Fix coverage test on PR and upload report

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -23,9 +22,9 @@ jobs:
...
+      run: go test -v -race -gcflags=-l -coverprofile=coverage.txt -covermode=atomic ./...
 
-    - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@v1
-      with:
-        file: ./coverage.txt
+    - name: Upload coverage report
+      uses: codecov/codecov-action@v4
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (b00f) [inline .github/workflows/coverage.yml:30]: What is this token? Do we need to set it?

Comment (Ja7ad) [reply .github/workflows/coverage.yml:30]: Codecov token, you can get in codecov configuration:

https://app.codecov.io/gh/pactus-project/pactus/

**Review by b00f — APPROVED** [review]

**Comment by b00f** [comment]
@Ja7ad please avoid using `fix` keyword for all PR titles. Thanks

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -23,9 +22,9 @@ jobs:
       uses: actions/checkout@v4
 
     - name: Test with coverage
-      run: go test -gcflags=-l -coverprofile=coverage.txt -covermode=atomic ./...
+      run: go test -v -race -gcflags=-l -coverprofile=coverage.txt -covermode=atomic ./...
```
Comment (b00f) [inline .github/workflows/coverage.yml]: We don't need to add race here. We have another action for race detection.

**Review by b00f — APPROVED** [review]

**Pull request merged**